stdlib/mexpr/cmp.mc

```diff
@@ -392,9 +392,29 @@ lang VariantTypeCmp = Cmp + VariantTypeAst
   | (TyVariant t1, TyVariant t2) -> mapCmp cmpType t1.constrs t2.constrs
 end
 
+lang ConTypeCmp = Cmp + ConTypeAst
+  sem cmpTypeH =
+  | (TyCon t1, TyCon t2) -> nameCmp t1.ident t2.ident
+end
+
 lang VarTypeCmp = Cmp + VarTypeAst
+  sem cmpTVar =
+  | (Unbound {ident = n}, Unbound {ident = m}) -> nameCmp n m
+  | (Link ty1, Link ty2) -> cmpType ty1 ty2
+  | (tv1, tv2) ->
+    subi (constructorTag tv1) (constructorTag tv2)
+
   sem cmpTypeH =
   | (TyVar t1, TyVar t2) -> nameCmp t1.ident t2.ident
+  | (TyFlex t1, TyFlex t2) -> cmpTVar (deref t1.contents, deref t2.contents)
```

Comment:
I think what I was referring to was the implementation of `eqType` for `TyVar` and `TyAll`, but I have since updated it to be correct.

You're right, I should resolve the links here as well.